Scaffold Adorsys Theme Plugin to mirror a classical moodle theme

freshystar opened the pull request:
The requested changes were met as follows:

- Deleted package.json and yarn.lock found at the root of the project
- Deleted adorsys_theme_v1/scss/style.scss
- Updated theme-ci.yaml
 - Documented setup, development, and Docker volume mounting in README.md.

freshystar committed:
Initialize and restructure Adorsys Theme plugin

• Flattened nested theme folder structure\n• Renamed plugin component and theme identifiers from 'adorsys-theme-v1' to 'adorsys_theme_v1' in version.php, config.php, and language files\n• Updated Docker Compose mount point in compose.yaml to reference the flattened theme path\n• Removed redundant nested directories and fixed language file naming

freshystar committed:
Add vertical sidebar layout overrides to Adorsys Theme

freshystar committed:
Enhance Adorsys Theme with interactive light/dark mode toggle

• Enabled 'class' based dark mode in Tailwind config\n• Included theme.js for toggle button in config.php\n• Added JavaScript to switch themes and persist preference

freshystar committed:
Add animated page transitions and initial page loaded class to Adorsys Theme

freshystar committed:
Revert SCSS to default Tailwind imports, retaining only header toggle dark mode feature

freshystar committed:
Refactor code structure for improved readability and maintainability

stephane-segning commented on .github/workflows/ci.yml:
Why did you removed the Rust CI?

stephane-segning commented on plugins/adorsys_theme_v1/config.php:
```diff
@@ -0,0 +1,14 @@
+defined('MOODLE_INTERNAL') || die();
+
+$THEME->name = 'adorsys_theme_v1';
+$THEME->parents = ['boost'];
```
Why is the parent boost? I think it's best to have the base as parent right? And customize it using tailwindcss and webpack

stephane-segning commented on plugins/adorsys_theme_v1/javascript/theme.js:
Never use a `.js` where you should use `.ts` instead

stephane-segning commented on plugins/adorsys_theme_v1/postcss.config.js:
This is not how to configure tailwindcss with psotcss. Use instead https://tailwindcss.com/docs/installation/using-postcss

stephane-segning commented on plugins/adorsys_theme_v1/scss/style.scss:
This is not how to configure tailwindcss v4 with psotcss. Use instead https://tailwindcss.com/docs/installation/using-postcss

stephane-segning commented on plugins/adorsys_theme_v1/style/all.css:
It makes no sense to vcs this file, since it's auto-generated

stephane-segning commented on plugins/adorsys_theme_v1/style/bundle.js:
Why do we have a `.js` file under style `style`?

stephane-segning commented on plugins/adorsys_theme_v1/tailwind.config.js:
https://tailwindcss.com/docs/installation/using-postcss

stephane-segning replied:
This file is not required in v4

stephane-segning commented on plugins/adorsys_theme_v1/webpack.config.js:
Change this and configure using typescript instead.

stephane-segning replied:
Also, read more about webpack. It'll be our main tool for developing themes, so better acquire knowledge. And it should look like this:

```typescript
import { Configuration } from "webpack";
// other imports ...

const config: Configuration = {
    entry: {
        main: "./src/index.ts"
    },
    devtool: "source-map",
    output: {
        clean: true,
        filename: pathData => {
            return pathData.chunk?.name === "runtime" || pathData.chunk?.name === "main"
                ? "js/[name].js"
                : "js/[name]-[fullhash].js";
        },
        path: path.resolve(__dirname, "data/common/resources"),
        chunkFilename: "js/chunks/[name]-[fullhash].js"
    },
    mode: "production",
    resolve: {
        extensions: [".ts"]
    },
    module: {
        rules: [
            {
                // Match `.js`, `.jsx`, `.ts` or `.tsx` files
                test: /\.[jt]sx?$/,
                loader: "esbuild-loader"
            },
            {
                test: /\.html$/,
                use: "html-loader"
            },
            {
                test: /\.(s|)[ca]ss$/,
                use: [
                    MiniCssExtractPlugin.loader,
                    "css-loader",
                    "postcss-loader",
                    "sass-loader"
                ]
            }
        ]
    },
    // if you need optimization,...
    optimization: {
        splitChunks: {
            filename: "js/split/[name]-[fullhash].js",
            minChunks: 1,
            minSize: 100_000
        },
        // if you need minimizer,...
        minimizer: [
            new EsbuildPlugin({
                target: "esnext",
                css: true
            }),
            new CssMinimizerPlugin()
        ],
        minimize: true,
        runtimeChunk: "single"
    },
    // if you need plugins,...
    plugins: [
        // if you need mini css,...
        new MiniCssExtractPlugin({
            filename: "css/[name].css",
            chunkFilename: "css/chunks/[id].css"
        }),
        // if you need copy,...
        new CopyPlugin({
            patterns: [
                {
                    from: path.resolve(__dirname, "../..", "assets-img"),
                    to: path.resolve(__dirname, "data/common/resources/img/")
                }
            ]
        })
    ]
};

export default config;
```

stephane-segning commented on plugins/adorsys_theme_v1/postcss.config.js:
Add [CssNano](https://cssnano.github.io/cssnano/docs/getting-started/) too

stephane-segning reviewed (changes requested):
Good work, but too much AI-inspired. 
1. We want to use taiwind 4 and webpack 5.
2. We will use TypeScript 5
3. We will have a src and dist folder, for source and destination code respectively. Please structure your code accordingly.
4. One other reason to use PHP themes for Moodle is to change and play with the layout.
5. Don't base your theme on boost too. Start it from the bottom.

freshystar committed:
feat:
-  Updated tailwind, webpack and typescript versions.
- created src and dist directory
-  changed the theme parent from boost to null [] inorder to be able to build from scratch and customize the theme
- Correctly configured tailwindcss with postcss
- Removed bundle.js and all.css
- Deleted tailwind.config.js
- Changed webpack.config.js to webpack.config.ts
- Added CssNano to postcss.config.js

freshystar committed:
feat: added layout files and edited lib.php

freshystar committed:
feat: added template/default.mustache file, updated config.php file to match moodle's expected layout ttypes

freshystar committed:
fix: Reduced settings gear, home icon, my course icon

freshystar committed:
fix: Error, requested file not found

freshystar committed:
fix: redundant arrows

freshystar committed:
fix: removed redundant arrows under the settings

freshystar committed:
feat: updated config.php

freshystar committed:
feat: updated templates, deleted javasrcipt folder, renamed columns2.php to columns.php, deleted columns1.php

freshystar committed:
fix: updated folder name from: cd plugins/adorsys-theme-v1 to cd plugins/adorsys_theme_v1 in theme-ci.yml

freshystar committed:
feat: updated theme-ci.yml

freshystar committed:
feat: updated theme-ci.yml

freshystar committed:
fix: The engine node is incompatible with this module. Expected version >=18. Got 16.20.2, upgraded from 16 to 18

freshystar committed:
refactor: streamline Webpack build and clean up redundant CSS entry

- Removed redundant SCSS entry from Webpack (imported via index.ts)
- Fixed CopyPlugin config to ensure proper asset output
- Ensured all theme assets are bundled and copied to the correct output path
- Cleaned up output directory structure (js/, style/, templates/, etc.)
- Updated tsconfig to match build setup

stephane-segning commented on .github/workflows/theme-ci.yml:
```diff
@@ -0,0 +1,29 @@
+          yarn build
+      - name: Verify build outputs
+        run: |
+          test -f plugins/adorsys_theme_v1/src/styles/main.scss && test -f plugins/adorsys_theme_v1/dist/js/main.js
```
Here you're testing `*.scss` considered a source file and `dist/js/main.js` considered a built file. Choose which kind of files to test

freshystar replied:
Ok sir

stephane-segning commented on package.json:
Why is this package existing?

freshystar replied:
Because you asked me to node a theme project, so while doing yarn init I got a package.json at the root of the theme project

stephane-segning replied:
Yes. Then remove it. You already have one at the plugin level

stephane-segning commented on plugins/gis-theme/adorsys_theme_v1/package.json:
```diff
@@ -0,0 +1,41 @@
+    "webpack-dev-server": "5.2.2"
+  },
+  "dependencies": {
+    "@tailwindcss/typography": "^0.5.16"
```
Why is the typo as normal dep?

freshystar replied:
Sir this was for a test, I will remove it

stephane-segning commented on plugins/gis-theme/adorsys_theme_v1/scss/style.scss:
Remove this file, since you're already using a `main.scss` somewhere else

freshystar replied:
ok sir

stephane-segning commented on yarn.lock:
Why this file?

freshystar replied:
I was using yarn, so while doing the yarn install a yarn.lock file was created. Do you suggest I add it to the .gitignore?

stephane-segning replied:
You're using yarn, ok. But here we're at project level. Why do you need this? Are you using yarn workspace?

freshystar committed:
fix: deleted package.json and yarn.lock found at the root, deleted scss/style.scss

leghadjeu-christian commented on plugins/gis-theme/adorsys_theme_v1/README.md:
```diff
@@ -0,0 +1,94 @@
+├── postcss.config.mjs        # PostCSS plugins (e.g., Tailwind, autoprefixer)
+│
+├── scss/
+│   └── style.scss            # Tailwind CSS entry
```
It looks good, please update your README to reflect the changes you made like the style.scss file you deleted for example.

freshystar replied:
ok

freshystar committed:
chore: update README.md to reflect project structure changes and include yarn.lock

stephane-segning commented:
@codex review this pr point by point

freshystar committed:
fix: update Node.js version requirement to >=18 and enhance block handling in layout files

freshystar committed:
fix: update CI workflow to disable cache and properly cache Yarn dependencies

freshystar committed:
fix: remove cache option from Node.js setup in CI workflow

freshystar committed:
fix: update CI workflow and package.json for improved Node.js handling and build process

freshystar committed:
fix: update README.md to clarify the purpose of lib.php in the project structure

stephane-segning commented on .github/workflows/theme-ci.yml:
```diff
@@ -0,0 +1,42 @@
+on:
+  push:
+    branches:
+      - adorsys-theme-v1
```
Don't use only this branch

stephane-segning commented:
@codex review this pr point by point

freshystar committed:
fix: remove branch restriction for push events in CI workflow

freshystar committed:
fix: update GitHub Actions to use latest versions of checkout, setup-node, and cache actions

stephane-segning commented on .github/workflows/theme-ci.yml:
```diff
@@ -0,0 +1,40 @@
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
Why not v5?

stephane-segning commented on .github/workflows/theme-ci.yml:
```diff
@@ -0,0 +1,40 @@
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
```
What's the latest version?

freshystar committed:
fix: update actions/checkout and actions/setup-node to latest versions

freshystar committed:
Merge branch 'main' into adorsys-theme-v1

freshystar committed:
docs: add installation guide for Adorsys Theme v1

freshystar committed:
Merge branch 'adorsys-theme-v1' of github.com:ADORSYS-GIS/moodle-plugin into adorsys-theme-v1

marcjazz commented on docs/guides/06-adorsys-theme_v1.md:
This is dos is end-user focus. And if that is the case then we don't need it on GitHub. How do I, as a developer, mount this plugin directly to the moodle plugin directory directly from the compose file ?

freshystar replied:
Sir for the theme name, I was asked to put "adorsys_theme_v1", so that's it's actual name.
As for the docs, I will update it. What I got from the ticket was that I should explain to  future users how to integrate it to their moodle stack, and I thought the future user was an "end-user", not a developer, I will fix that.

marcjazz commented on plugins/gis-theme/adorsys_theme_v1/README.md:
````diff
@@ -0,0 +1,91 @@
+```
+Then restart:
+```bash
+docker compose up -d
````
I did this it didn't work. Do you know why?
...May be you could also add plugin download step here too.

freshystar replied:
Sir, I don't get you. What do you mean by " I should add plugin download step here" ?

marcjazz reviewed (changes requested):
Nice job but I wasn't able to set the theme as my current theme after uploading it to moodle. 

Also have you consider the possibility of given this theme an actual name? In that case, you will have to update each and every file of your project that hard coded the theme name...

stephane-segning commented:
@codex review this

freshystar committed:
refactor: remove theme plugin volume from docker-compose and update installation guide
